go/common/apivalidator/validate_query.go

```diff
@@ -0,0 +1,18 @@
+package apivalidator 
+
+import (
+	"fmt"
+)
+
+
+
+func ValidateQueryInput(sortOrder *SortOrder) error {
+	if sortOrder == nil {
+		return nil
+	}
+	if *sortOrder != SortOrderAscendant && *sortOrder != SortOrderDescendant {
+		return fmt.Errorf(ErrInvalidSortOrder,
```

Comment:
@OlfaBensoussia we added a new way of handling errors and I would suggest adapting this to the existing error handling method can be better for consistency reasons.